File: Solution with Bitwise Operations.cs

using System;

public class Solution
{
    private static readonly int[] ID_RANGE = { 1, 100 };

    public int MaxStudentsOnBench(int[][] students)
    {
        int maxStudentsOnBench = 0;
        int[] frequencyStudentsPerBench = new int[ID_RANGE[1] + 1];
        long[,] bitStampForSeatedStudentsOnBench = new long[ID_RANGE[1] + 1, 2];

        foreach (int[] current in students)
        {
            int studentID = current[0];
            int benchID = current[1];

            int index = studentID / (1 + ID_RANGE[1] / 2);
            int bitShifts = studentID % (1 + ID_RANGE[1] / 2);
            long bitStampStudent = 1L << bitShifts;

            if (!IsSeated(bitStampForSeatedStudentsOnBench[benchID, index], bitStampStudent))
            {
                ++frequencyStudentsPerBench[benchID];
                bitStampForSeatedStudentsOnBench[benchID, index] ^= bitStampStudent;
                maxStudentsOnBench = Math.Max(maxStudentsOnBench, frequencyStudentsPerBench[benchID]);
            }
        }

        return maxStudentsOnBench;
    }

    private bool IsSeated(long bitStampForSeatedStudentsOnBench, long bitStampStudent)
    {
        return (bitStampForSeatedStudentsOnBench & bitStampStudent) != 0;
    }
}

File: Solution with Bitwise Operations.cpp

#include <array>
#include <vector>
#include <algorithm>
using namespace std;

class Solution {

    static constexpr array<int, 2> ID_RANGE = { 1, 100 };

public:
    int maxStudentsOnBench(const vector<vector<int>>& students) const {

        int maxStudentsOnBench = 0;
        array<int, ID_RANGE[1] + 1> frequencyStudentsPerBench{};
        array<array<long long, 2>, ID_RANGE[1] + 1>  bitStampForSeatedStudentsOnBench{};

        for (const auto& current : students) {
            int studentID = current[0];
            int benchID = current[1];

            int index = studentID / (1 + ID_RANGE[1] / 2);
            int bitShifts = studentID % (1 + ID_RANGE[1] / 2);
            long long bitStampStudent = 1LL << bitShifts;

            if (!isSeated(bitStampForSeatedStudentsOnBench[benchID][index], bitStampStudent)) {
                ++frequencyStudentsPerBench[benchID];
                bitStampForSeatedStudentsOnBench[benchID][index] ^= bitStampStudent;
                maxStudentsOnBench = max(maxStudentsOnBench, frequencyStudentsPerBench[benchID]);
            }
        }

        return maxStudentsOnBench;
    }

private:
    bool isSeated(long long bitStampForSeatedStudentsOnBench, long long bitStampStudent) const {
        return (bitStampForSeatedStudentsOnBench & bitStampStudent) != 0;
    }
};
